### PR TITLE
spago init template based on Spec

### PR DESCRIPTION
**Pull request by fsoikin**
### Description of the change

It occurred to me that, since [we're recommending `spec-node` in the docs](https://github.com/purescript/spago/pull/1263#pullrequestreview-2221076069) now, it would make sense to include it in the `spago init` template. So the user doesn't have to jump through extra hoops.

I understand this may not be 100% desirable because it introduced extra moving parts. Feel free to throw away. Just an idea.

### Checklist:

- [x] Added the change to the "Unreleased" section of the changelog
- ~[ ] Added some example of the new feature to the `README`~
- [x] Added a test for the contribution (if applicable)

### CODE REVIEW

File: test/Prelude.purs [hunk 1]
```diff
@@ -26,7 +26,7 @@ import Spago.Command.Init as Init
...
... bottom, catchError, clamp, compare, comparing, compose, conj, const, degree, die, die', discard, disj, div, either, eq, except, flap, flip, foldMap, foldl, for, forWithIndex, for_, fromMaybe, fst, gcd, genericShow, hush, identity, ifM, indent, indent2, isJust, isLeft, isNothing, isPrefix, isRight, join, justOrDieWith, justOrDieWith', lcm, liftA1, liftAff, liftEffect, liftM1, lmap, local, logDebug, logError, logInfo, logSuccess, logWarn, map, max, maybe, mempty, min, mkTemp, mkTemp', mod, mul, negate, not, notEq, on, one, or, otherwise, output, parTraverseSpago, parallelise, parseJson, parseLenientVersion, parseUrl, parseYaml, partition, partitionEithers, partitionMap, printJson, printYaml, pure, recip, rightOrDieWith, rightOrDieWith', rmap, runSpago, shaToHex, show, snd, sub, toDoc, top, traverse, try, typoSuggestions, unit, unless, unlessM, unsafeFromJust, unsafeFromRight, unsafeLog, unsafeStringify, unsafeThrow, unwrap, void, when, whenM, withBackoff', withForwardSlashes, zero, (#), ($), ($>), (&&), (*), (*>), (+), (-), (..), (/), (/=), (/\), (:), (<), (<#>), (<$), (<$>), (<*), (<*>), (<<<), (<=), (<=<), (<>), (<@>), (<|>), (=<<), (==), (>), (>=), (>=>), (>>=), (>>>), (||)) as X
```
Comment (f-f) [inline test/Prelude.purs:29]: Can we rollback this line? It's just confusing to have this huge import like this

**Review by f-f — APPROVED** [review]
Ah yeah, this makes sense!

File: test/Prelude.purs (hunk 1 above)
Comment (fsoikin) [reply test/Prelude.purs:29]: Oh yeah, sorry, that's accidental, from IDE support.

**Comment by f-f** [comment]
I think this is a good change, but after all the updates to the tests the CI times have ballooned - they are now 2x slower than the current trunk.

That's a bit of a deal breaker, so let's think about an alternative approach; the no-hassle one is to add some `--no-tests` flag to `spago init`, to skip the `spec-node` addition and use the current setup - we could then use this flag for most of the tests, and only add the new testing setup to a few of them, while still keeping it as the default for spago.
However, that would not be extensible. A more general approach could be to introduce templates for `spago init` - then we could have a `basic` template that would contain the bare minimum (basically as it is now), and a `default` template that would add `spec-node`. This would open the door to more complex things, e.g. a `halogen` template, and so on.
Does this make any sense?

**Comment by fsoikin** [comment]
Yes, I agree, this became quite unmanageable. I was hoping it would literally just be changing the template, but it turns out all the tests depend on it, and it's quite a ball of yarn. 

I do like the idea of multiple templates. Other tools have similar things (e.g. dotnet) and they're very popular. But I don't think it is critical for the current milestone. So I guess we should just table this change. Or throw it away altogether.

**Comment by f-f** [comment]
Yeah let's not move forward with this as is, and I agree with you that it's not critical for the current milestone. I have reopened #573 to track this effort.

**Pull request closed without merging**